src/util/rocksdb_util/properties.rs

```diff
@@ -940,4 +947,60 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_range_properties_with_blob_index() {
+        let cases = [
+            ("a", 0),
+            // handle "a": size(size = 1, offset = 1),keys(1,1)
+            ("b", PROP_SIZE_INDEX_DISTANCE / 8),
+            ("c", PROP_SIZE_INDEX_DISTANCE / 4),
+            ("d", PROP_SIZE_INDEX_DISTANCE / 2),
+            ("e", PROP_SIZE_INDEX_DISTANCE / 8),
+            // handle "e": size(size = DISTANCE + 4, offset = DISTANCE + 5),keys(4,5)
+            ("f", PROP_SIZE_INDEX_DISTANCE / 4),
+            ("g", PROP_SIZE_INDEX_DISTANCE / 2),
+            ("h", PROP_SIZE_INDEX_DISTANCE / 8),
+            ("i", PROP_SIZE_INDEX_DISTANCE / 4),
+            // handle "i": size(size = DISTANCE / 8 * 9 + 4, offset = DISTANCE / 8 * 17 + 9),keys(4,5)
+            ("j", PROP_SIZE_INDEX_DISTANCE / 2),
+            ("k", PROP_SIZE_INDEX_DISTANCE / 2),
+            // handle "k": size(size = DISTANCE + 2, offset = DISTANCE / 8 * 25 + 11),keys(2,11)
+        ];
+
+        let handles = ["a", "e", "i", "k"];
+
+        let mut rng = rand::thread_rng();
+        let mut collector = RangePropertiesCollector::default();
+        let mut extra_value_size: u64 = 0;
+        for &(k, vlen) in &cases {
+            if handles.contains(&k) || rng.gen_range(0, 2) == 0 {
+                let v = vec![0; vlen as usize - extra_value_size as usize];
+                extra_value_size = 0;
+                collector.add(k.as_bytes(), &v, DBEntryType::Put, 0, 0);
+            } else {
+                let mut blob_index = TitanBlobIndex::default();
+                blob_index.blob_size = vlen - extra_value_size;
+                let v = blob_index.encode();
+                extra_value_size = v.len() as u64;
```

Comment:
I'm confused by the way of calculating `extra_value_size`, could you give some comments?